doc/client.md

```diff
@@ -0,0 +1,395 @@
+### Interface: client
+
+#### Client()
+
+HTTP Client interface for Impress Application Server
+
+
+#### Client.prototype.constructor(application, req, res)
+
+- `application`: [`<Application>`] instance of impress.Application
+- `req`: [`<http.IncomingMessage>`]
+- `res`: [`<http.ServerResponse>`]
+
+Client constructor
+
+
+#### Client.prototype.parseCookies()
+
+Convert cookie string from request's headers to `client.cookies` object
+
+_Example:_
+```js
+ 'name=value; path=/' => { name: 'value', path: '/' }
+```
+
+
+#### Client.prototype.setCookie(name, value, host, httpOnly)
+
+- `name`: [`<string>`] cookie name
+- `value`: [`<string>`] cookie value
+- `host`: [`<string>`] host name, optional, default: client.req.headers.host
+- `httpOnly`: [`<boolean>`] HttpOnly cookie modifier, optional, default: true
+
+Set cookie by adding it to `client.preparedCookies`
+
+
+#### Client.prototype.deleteCookie(name, host)
+
+- `name`: [`<string>`] cookie name
+- `host`: [`<string>`] optional, default: client.req.headers.host
+
+Delete cookie
+
+
+#### Client.prototype.sendCookie()
+
+Set all prepared cookies if headers have not been sent yet
+
+
+#### Client.prototype.proxy(hostname, port, path)
+
+- `hostname`: [`<string>`] forward request to host name or IP address
+- `port`: [`<number>`] request port
+- `path`: [`<string>`] request path
+
+Route request to external HTTP server
+
+
+#### Client.prototype.dispatch()
+
+Check application firewall access, in case if access is allowed
+
```

Comment:
@SemenchenkoVitaliy why did `metadoc` add an empty line here?

Reply:
This is the title. It should be a one-liner as specified [here](https://github.com/metarhia/metadoc/blob/master/test/example.js#L5). All other lines before parameters will be moved to their separate section - description.

Reply:
Have not got completely from this [example](https://github.com/metarhia/metadoc/blob/3d8bae1763a253c9c876f74297d8ee0de4e212b9/test/example.js#L5) that first line as a title in multi-line description is obligatory to be generated correctly.

Reply:
@belochub @SemenchenkoVitaliy I've fixed this and other similar comments, haven't squashed commits yet so that you could quickly see changes in fixup.